xdsl/interpreter.py

```diff
@@ -331,8 +388,43 @@ def interpreter_assert(self, condition: bool, message: str | None = None):
             raise InterpretationError(f"AssertionError: ({self._ctx})({message})")
 
 
+PythonValues: TypeAlias = tuple[Any, ...]
+
+
+class ReturnedValues(NamedTuple):
+    values: PythonValues
+
+
+TerminatorValue: TypeAlias = ReturnedValues
+
+
+class OpImplResult:
```

Comment:
Can you document this, and make it a dataclass?

Reply:
made it a NamedTuple instead

Reply:
Makes sense!